src/pkcs11/framework-pkcs15.c

```diff
@@ -5310,7 +5310,16 @@ struct sc_pkcs11_object_ops pkcs15_profile_ops = {
 static void
 pkcs15_skey_release(void *object)
 {
-	__pkcs15_release_object((struct pkcs15_any_object *) object);
+	struct pkcs15_skey_object *skey = (struct pkcs15_skey_object *)object;
+	struct sc_pkcs15_skey_info *skey_info = skey->info;
+	struct sc_pkcs15_object *p15obj = skey->skey_p15obj;
+
+	if (__pkcs15_release_object((struct pkcs15_any_object *)skey) == 0) {
+		if (p15obj->session_object) {
+			sc_pkcs15_free_skey_info(skey_info);
+			free(p15obj);
```

Comment:
Why is this step needed? skey_p15obj is a define for base.p15_object just like, for example, prv_p15obj, but none of the other are calling `free` on base.p15_object in their release functions...

Reply:
The answer here is basically the same as in the above comment -- because it is not stored in the p15card's directory structure so it can not be freed from there.

Initially I considered using some reference counter, but unfortunately the `sc_pkcs15_object` does not have reference counter so I did it this ugly way.